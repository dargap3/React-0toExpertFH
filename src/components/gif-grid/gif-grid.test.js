import { shallow } from "enzyme";
import { useFetchGifs } from "../../hooks/useFetchGifs";

import GifGrid from "./gif-grid";

jest.mock("../../hooks/useFetchGifs");

describe('<GifGrid /> tests', () => {
  const category = "I'm a category";

  test('should show correctly', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    });

    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
  });

  test('should show items when you charge images useFetchGifs', () => {
    const gifs = [{
      id: 'ABC',
      url: 'https://localhost/cualquier/cosa.jpg',
      title: 'Cualquier cosa'
    },
    {
      id: 'ABdC',
      url: 'https://localhost/cualquier/cosa.jpg',
      title: 'Cualquier cosa'
    }];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    });
    
    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });
  
});
