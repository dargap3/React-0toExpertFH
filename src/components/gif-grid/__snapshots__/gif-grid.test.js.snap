// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifGrid /> tests should show correctly 1`] = `
<Fragment>
  <h3
    className="animate__fadeIn"
  >
    I'm a category
  </h3>
  <p
    className="animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
